esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3004,6 +3005,23 @@ void WsWuHelpers::submitWsWorkunit(IEspContext& context, IConstWorkUnit* cw, con
     const char *paramXml, IArrayOf<IConstNamedValue> *variables, IArrayOf<IConstNamedValue> *debugs, IArrayOf<IConstApplicationValue> *applications)
 {
     ensureWsWorkunitAccess(context, *cw, SecAccess_Write);
+
+#ifndef _NO_LDAP
+    CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(context.querySecManager());
```

Comment:
Additionally, it would be better if at startup, if checkViewPermissions is set, you go ahead and querySecManager(), saving in a member variable, and throwing if no LDAP security manager is found. This eliminate many conditional tests done at runtime.